src/primitives/transaction.h

```diff
@@ -372,6 +372,22 @@ class CTransaction : public CBaseTransaction
     std::string ToString() const;
 };
 
+class CLicenseTransaction : public CTransaction
+{
+	enum class ActionType {
+		NEW, MODIFICATION, REVOCATION
+	};
+
+	using CTransaction::CTransaction;
+
+public:
+	std::string ToString() const;
+
+private:
+	ActionType type;
+	uint64_t assignedHashrate;
```

Comment:
Also a licensed address should be a member of this class.

Reply:
Done